jsapp/js/account/plan.component.tsx

```diff
@@ -0,0 +1,442 @@
+            listItems.push({icon: iconBool, item: itemName});
+          }
+        }
+      })
+    );
+    return listItems;
+  };
+
+  const checkMetaFeatures = () => {
+    if (state.products.length > 0) {
+      filterPrices().map((price) =>
+        Object.keys(price.metadata).map((featureItem: string) => {
+          if (
+            featureItem.includes(`feature_support_`) ||
+            featureItem.includes(`feature_advanced_`) ||
+            featureItem.includes(`feature_addon_`)
+          ) {
+            setShowExpand(true);
+          } else {
+            setShowExpand(false);
+          }
+        })
+      );
+    }
+  };
+  if (!state.products.length) return null;
+  return (
+    <div className={styles.accountPlan}>
+      <div className={styles.plansSection}>
+        <form className={styles.intervalToggle}>
+          <input
+            type='radio'
+            id='switch_left'
+            name='switchToggle'
+            value='year'
+            onChange={() => dispatch({type: 'year'})}
+            checked={!state.filterToggle}
+          />
+          <label htmlFor='switch_left'>{t('Annual')}</label>
+
+          <input
+            type='radio'
+            id='switch_right'
+            name='switchToggle'
+            value='month'
+            onChange={() => dispatch({type: 'month'})}
+            checked={state.filterToggle}
+          />
+          <label htmlFor='switch_right'> {t('Monthly')}</label>
+        </form>
+        <div
+          className={styles.currentPlan}
+          style={{
+            gridRow: 0,
+            gridColumn: 1 + filterPrices().findIndex(isSubscribedProduct),
+            display:
+              filterPrices().findIndex(isSubscribedProduct) >= 0 ? '' : 'none',
+          }}
+        >
+          {t('your plan')}
+        </div>
+        {filterPrices().map((price: Price, i: number) => (
+          <div className={styles.planContainer} key={i}>
+            <h1 className={styles.priceName}> {price.name} </h1>
+            <div className={styles.priceTitle}>
+              {typeof price.prices.human_readable_price === 'string' &&
+                (price.prices.human_readable_price.includes('$0.00')
+                  ? t('Free')
+                  : price.prices.human_readable_price)}
+            </div>
+
+            <ul>
+              {Object.keys(price.metadata).map(
+                (featureItem: string) =>
+                  featureItem.includes('feature_list_') && (
+                    <li key={featureItem}>
+                      <div className={styles.iconContainer}>
+                        <Icon
+                          name='check'
+                          size='m'
+                          classNames={
+                            price.name === 'Professional plan'
+                              ? [styles.tealCheck]
+                              : [styles.stormCheck]
+                          }
+                        />
```

Comment:
Yeah I think it would be nice to pass in color as a parameter and if one isn't provided we stick with the default. I see it being useful in other scenarios as well. I will add as a to do on my list for improvements but not blockers